job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java

```diff
@@ -0,0 +1,639 @@
+import org.joda.time.DateTime;
+
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.UDTValue;
+import com.datastax.driver.core.UserType;
+import com.google.common.collect.Sets;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import rx.Completable;
+import rx.Observable;
+import rx.Single;
+import rx.functions.Action0;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+import rx.subjects.PublishSubject;
+
+/**
+ * @author jsanda
+ */
+public class SchedulerImpl implements Scheduler {
+
+    private static Logger logger = Logger.getLogger(SchedulerImpl.class);
+
+    private Map<String, Func1<JobDetails, Completable>> jobFactories;
+
+    private ScheduledExecutorService tickExecutor;
+
+    private rx.Scheduler tickScheduler;
+
+    private ExecutorService queueExecutor;
+
+    private rx.Scheduler queueScheduler;
+
+    private ExecutorService queryExecutor;
+
+    private rx.Scheduler queryScheduler;
+
+    private RxSession session;
+
+    private PreparedStatement insertJob;
+
+    private PreparedStatement insertScheduledJob;
+
+    private PreparedStatement findScheduledJobs;
+
+    private PreparedStatement deleteScheduledJobs;
+
+    private PreparedStatement findFinishedJobs;
+
+    private PreparedStatement deleteFinishedJobs;
+
+    private PreparedStatement updateJobToFinished;
+
+    private PreparedStatement findJob;
+
+    private PreparedStatement addActiveTimeSlice;
+
+    private PreparedStatement findActiveTimeSlices;
+
+    private PreparedStatement deleteActiveTimeSlice;
+
+    private LockManager lockManager;
+
+    private ConfigurationService configurationService;
+
+    private boolean running;
+
+    private AtomicInteger ticks = new AtomicInteger();
+
+    /**
+     * Test hook. See {@link #setTimeSlicesSubject(PublishSubject)}.
+     */
+    private Optional<PublishSubject<Date>> finishedTimeSlices;
+
+    private Optional<PublishSubject<JobDetails>> jobFinished;
+
+    public SchedulerImpl(RxSession session) {
+        this.session = session;
+        jobFactories = new HashMap<>();
+        tickExecutor = Executors.newScheduledThreadPool(1,
+                new ThreadFactoryBuilder().setNameFormat("ticker-pool-%d").build());
+        queueExecutor = Executors.newSingleThreadExecutor(
+                new ThreadFactoryBuilder().setNameFormat("job-queue-pool-%d").build());
+        tickScheduler = Schedulers.from(tickExecutor);
+        queueScheduler = Schedulers.from(queueExecutor);
+
+        queryExecutor = Executors.newFixedThreadPool(4,
+                new ThreadFactoryBuilder().setNameFormat("query-thread-pool-%d").build());
+        queryScheduler = Schedulers.from(queryExecutor);
+
+        lockManager = new LockManager(session);
+
+        insertJob = session.getSession().prepare(
+                "INSERT INTO jobs (id, type, name, params, trigger) VALUES (?, ?, ?, ?, ?)");
```

Comment:
No, these CQL statements should stay in the job-scheduler module. The job-scheduler module does not and will not depend on metrics-core-service. It may be the other way around where metrics-core-service depends on job-scheduler. Furthermore, there is nothing inherently specific to metrics data access in the job scheduler stuff, so pushing the CQL related code into metrics-core-service would make it somewhat convoluted IMO.